File: app/src/main/java/de/bikebean/app/db/settings/settings/location_settings/Lat.java
package de.bikebean.app.db.settings.settings.location_settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.settings.LocationSetting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.state.State;

public class Lat extends LocationSetting {

    private static final @NonNull State.KEY key = State.KEY.LAT;

    public Lat(double lat, final @NonNull Sms sms) {
        super(lat, sms, key);
    }

    public Lat() {
        super(key);
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/location_settings/Lat.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings.location_settings

import de.bikebean.app.db.settings.settings.add_to_list_settings.LocationSetting
import de.bikebean.app.db.settings.settings.add_to_list_settings.WappState
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.ui.drawer.status.location.ResponseBody

class Lat : LocationSetting {

    constructor(
            responseBody: ResponseBody,
            wappState: WappState
    ) : super(responseBody.location.lat, wappState, key)

    constructor() : super(key)

    companion object {
        private val key = KEY.LAT
    }

}
